### PR TITLE
Fixes around ancestor black lists

### PR DESCRIPTION
**Pull request by pierrewillenbrockdfki**
Seemed to work in my limited testing, let's see how it works on everything

### CODE REVIEW

File: bin/deb_local [hunk 1]
```diff
@@ -377,13 +377,13 @@ while pkgs_added
             next
         end
 
-        if ancestor_blacklist_patterns.select do |pat|
+        if !(ancestor_blacklist_patterns.select do |pat|
```
Comment (2maz) [inline bin/deb_local]: Not very readable, better use a local variable to cache the results.

File: lib/rock/packaging/debian.rb [hunk 2]
```diff
@@ -250,11 +250,7 @@ def update_osdeps_list(pkginfo, file, selected_platform)
                     pkg_name = pkginfo
                     release_name, is_osdep = native_dependency_name(pkg_name, selected_platform)
                     Packager.debug "Native dependency of ruby package: '#{pkg_name}' -- #{release_name}, is available as osdep: #{is_osdep}"
-                    if is_osdep
-                        dependency_debian_name = release_name
-                    else
-                        dependency_debian_name = debian_ruby_name(pkg_name)
-                    end
+                    dependency_debian_name = release_name
```
Comment (2maz) [inline lib/rock/packaging/debian.rb:253]: To increase consistency does it make sense to retrieve the pkginfo when string is given, and always use the else branch?

Comment (pierrewillenbrockdfki) [reply lib/rock/packaging/debian.rb:253]: A string is only provided if no pkginfo could be found by the calling code, so this code would not be able to find a pkginfo either. That probably happens for ruby packages that are not tracked by autoproj and do not exist in the operating system, and only when requested explicitly.

File: bin/deb_local (hunk 1 above)
Comment (pierrewillenbrockdfki) [reply bin/deb_local]: Just learned about any?. I would thing replacing !(array.select {} .empty?) by array.any? {} is better?

Comment (2maz) [reply bin/deb_local]: Yes, that should do.

**Pull request merged**